### ISSUE TITLE
Quality loss

### ISSUE THREAD

**Issue by LukaEHB** [leadedge/SpoutProcessing#6]
Hello there,

I am using processing to send a texture to Resolume. The resolution in Resolume seems to be the same but there is a lot of quality loss. It is not as sharp as the processing output.
What can be the cause of this?

```
import spout.*;
Spout spout;

// Screen
int fps = 30;

void setup() {
  size(1080, 1080, P2D);
  frameRate(fps);

  spout = new Spout(this);
  spout.createSender("Spout Processing"); 
}

void draw() {
  noFill();
  stroke(255, 102, 0);
  strokeWeight(5);
  curve(0,0, 55,55,200,400, 600,800);
  spout.sendTexture(); //mySender.SendTexture(mesh.texture, GL_TEXTURE_2D, rwidth, rheight);
}
```

It's hard to see on a screenshot maybe. But it loses a lot of crisp.
On a big sketch, the difference is really visible.

Processing:
![processing](https://user-images.githubusercontent.com/45628865/79041960-1bfca400-7bf4-11ea-9183-a5007569f298.png)

Resolume:
![resolume](https://user-images.githubusercontent.com/45628865/79041964-1f902b00-7bf4-11ea-92b2-0cf32e1761ce.png)

**Comment by leadedge** [comment]
Yes I can certainly see the difference. I notice :

size(1080, 1080, P2D);

change that to :

size(1080, 1080, P3D);

See if that makes a difference. If not add :

textureMode(NORMAL);

Otherwise I can suggest some tests to narrow it down.

**Comment by leadedge** [comment]
I had some time so made some tests. 

I find that P2D/P3D makes no difference for your application, but P3D is the default for Spout applications unless there is a particular requirement for P2D.

I believe the problem is due to resolution difference between the Processing sender and Resolume, combined with the anti-aliasing of the Resolume rendering.

If Arena is set for a composition resolution of 1280x720, the Processing sketch resolution of 1920x1080 is too large for Resolume to display the entire Spout output, so the clip resolution has to be reduced to 1280x720 as well. In that case, if you go full screen (1920x1080), Resolume renders the 1280x720 source the best it can with anti-aliasing. In that situation I can reproduce your observations (image 1).

If I set Arena composition resolution to 1920x1080, and leave the Spout source clip at 1920x1080, the result is almost the same aside from anti-aliasing at one level (image 2).

If I use the Spout demo Receiver, which performs very simple rendering, the result is exactly the same as the Processing sender (image 3). This shows that there is no quality loss for sharing textures.

Generally anti-aliasing is desirable to make clean edges without jagginess. I do not know of any way to disable this for Resolume output.

Could you make sure the composition output is set to 1920x1080 to see whether the result is improved as shown in image 2.

![Processing-compare](https://user-images.githubusercontent.com/7351229/79063952-2b5b1a80-7ce4-11ea-9906-c78310ac7166.png)

**Comment by LukaEHB** [comment]
> nd that P2D/P3D
Thanks for your help already!
But my Arena composition resolution was already set on 1920x1080. My processing sketch is a 1080x1080. And I did try to change it to P3D with no difference in Resolume. Alsobut the drawings looks different in processing when going for 3D. So I prefer the P2D.

**Comment by LukaEHB** [comment]
Thanks for your help already!
But my Arena composition resolution was already set on 1920x1080. My processing sketch is a 1080x1080. 

I did try to change it to P3D with no difference in Resolume. Also, the drawings in some sketches look different in processing when going for 3D. So I prefer the P2D.

Tested the "textureMode(NORMAL);" but no luck either :/.

I opened the spout demo receiver sketch, and indeed it looks very similar to the original.
For some reason, the receiver in Resolume always does the anti-aliasing...

1. Resolume - 2. Processing sketch - 3. Demo Receiver sketch
![resolume](https://user-images.githubusercontent.com/45628865/79224006-aa725f00-7e5a-11ea-9550-a89c735fe8a7.png)

**Comment by leadedge** [comment]
I can't reproduce your findings.

It is possible that this is caused by driver settings. First what is your hardware? Run the Spout demo receiver. Help > Diagnostics. Click copy and paste here.

For NVIDIA hardware, open the NVIDIA control panel.
Manage 3D settings > Global settings
Set all these off
Image Sharpening
Antialiasing - FXAA
Antialiasing - Gamma correction (on)
Antialiasing - Mode Application controlled
Antialiasing - Transparency

Also I upgraded Arena to 7.12 rev 68973. Now with Arena running, the sketch window is white and can only be closed with the stop button. The sketch gives me a message : "The sketch has been automatically resized to fit the screen resolution". This also happens if I set the size to 1280x720. Arena 6 is OK. I don't have time at the moment to re-install and earlier version of Arena 7 but will do so.
